airflow/providers/google/cloud/hooks/dataflow.py

```diff
@@ -938,6 +937,110 @@ def launch_job_with_flex_template(
+        :param yaml_pipeline_file: Path to a file defining the YAML pipeline to run.
+            Must be a local file or a URL beginning with 'gs://'.
+        :param append_job_name: Set to True if a unique suffix has to be appended to the `job_name`.
+        :param jinja_variables: A dictionary of Jinja2 variables to be used in reifying the yaml pipeline file.
+        :param options: Additional gcloud or Beam job parameters.
+            It must be a dictionary with the keys matching the optional flag names in gcloud.
+            The list of supported flags can be found at: `https://cloud.google.com/sdk/gcloud/reference/dataflow/yaml/run`.
+            Note that if a flag does not require a value, then its dictionary value must be either True or None.
+            For example, the `--log-http` flag can be passed as {'log-http': True}.
+        :param project_id: The ID of the GCP project that owns the job.
+        :param location: Region ID of the job's regional endpoint. Defaults to 'us-central1'.
+        :param on_new_job_callback: Callback function that passes the job to the operator once known.
+        :return: Job ID.
+        """
+        job_name = self.build_dataflow_job_name(job_name, append_job_name)
+        cmd = self._build_yaml_gcloud_command(
+            job_name=job_name,
+            yaml_pipeline_file=yaml_pipeline_file,
+            project_id=project_id,
+            region=location,
+            options=options,
+            jinja_variables=jinja_variables,
+        )
+        job_id = self._create_dataflow_job_with_gcloud(cmd=cmd)
+        return job_id
+
+    def _build_yaml_gcloud_command(
+        self,
+        job_name: str,
+        yaml_pipeline_file: str,
+        project_id: str,
+        region: str,
+        options: dict[str, Any] | None,
+        jinja_variables: dict[str, str] | None,
+    ) -> list[str]:
+        gcp_flags = {
+            "yaml-pipeline-file": yaml_pipeline_file,
+            "project": project_id,
+            "format": "value(job.id)",
+            "region": region,
+        }
+
+        if jinja_variables:
+            gcp_flags["jinja-variables"] = json.dumps(jinja_variables)
+
+        if options:
+            gcp_flags.update(options)
+
+        if self.impersonation_chain:
+            if isinstance(self.impersonation_chain, str):
+                impersonation_account = self.impersonation_chain
+            elif len(self.impersonation_chain) == 1:
+                impersonation_account = self.impersonation_chain[0]
+            else:
+                raise AirflowException(
+                    "Chained list of accounts is not supported, please specify only one service account."
+                )
+            gcp_flags.update({"impersonate-service-account": impersonation_account})
+
+        return [
+            "gcloud",
+            "dataflow",
+            "yaml",
+            "run",
+            job_name,
+            *(beam_options_to_args(gcp_flags)),
+        ]
```

Comment:
This logic is taken from `DataflowHook.start_sql_job`. Maybe you could extract it to a utility function that will be used by both? (with the required adjustments)
Then you could test it separately

Reply:
Thanks for the suggestion. Please take a look at the update.